opendevin/controller/agent_controller.py

```diff
@@ -209,7 +208,8 @@ async def start_delegate(self, action: AgentDelegateAction):
         state = State(
             inputs=action.inputs or {},
             iteration=0,
-            max_iterations=self.state.max_iterations,
+            max_iterations=min(self.state.max_iterations, self.state.global_max_iterations - self.state.iteration),
```

Comment:
this won't work because `self.state.iteration` only records the current agent's number of iterations, not the global number of iterations.

We need a new variable like `global_iteration`.